compiler/include/misc.h

```diff
@@ -87,10 +87,16 @@
 
 class BaseAST;
 
+enum GpuCodegenType {
+  GPU_CG_CUDA,
+  GPU_CG_AMD,
```

Comment:
We're mixing terminology here. `GPU_CG_CUDA` uses the middleware layer (CUDA vs HIP) whereas `GPU_CG_AMD` uses the vendor (NVIDIA vs AMD).

In reality this enum affects both (in one place it controls whether we pass `-x cuda` or `-x hip` to `clang` and in another place it influences things like what target triple we use and what toolchain we use to package the fatbin file).

I think I'd actually spell it out as:

```
GPU_CG_NVIDIA_CUDA
GPU_CG_AMD_HIP
```

Since in theory we could have `NVIDIA_HIP` (although we're unlikely to support it).